src/datasets/io/sql.py

```diff
@@ -48,7 +48,7 @@ def read(self):
 
         # Build dataset for splits
         dataset = self.builder.as_dataset(
-            split="train", verification_mode=verification_mode, in_memory=self.keep_in_memory
+            split=self.split, verification_mode=verification_mode, in_memory=self.keep_in_memory
```

Comment:
I would not add the `split` parameter to the `SqlDatasetReader` because it is unrelated to the objective of this PR.